### ISSUE TITLE
ContraWR / DeepSleepNet / EEGMiner: confusing error on batch=1 forward (BatchNorm in default train mode)

### ISSUE THREAD

**Issue by bruAristimunha** [braindecode/braindecode#1025]
## Summary

Several braindecode models raise a confusing `ValueError` when running a forward pass with `batch_size == 1` because the model is in PyTorch's default **train** mode and an internal BatchNorm refuses a single-sample batch.  Setting `.eval()` after construction fixes it.

## Reproducer

```python
import torch
from braindecode.models.util import models_dict

for name in ["ContraWR", "DeepSleepNet", "EEGMiner"]:
    Cls = models_dict[name]
    model = Cls(n_chans=22, n_outputs=4, n_times=1000, sfreq=200.0)
    x = torch.randn(1, 22, 1000)

    # 1) Default (train) mode — fails.
    try:
        with torch.no_grad():
            model(x)
    except ValueError as e:
        print(f"{name}: train mode → {e}")

    # 2) Eval mode — passes.
    model.eval()
    with torch.no_grad():
        out = model(x)
    print(f"{name}: eval mode  → output {tuple(out.shape)}")
```

Output:
```
ContraWR: train mode → Expected more than 1 value per channel when training, got input size torch.Size([1, 256, 1, 1])
ContraWR: eval mode  → output (1, 4)
DeepSleepNet: train mode → Expected more than 1 value per channel when training, got input size torch.Size([1, 1024])
DeepSleepNet: eval mode  → output (1, 4)
EEGMiner: train mode → Expected more than 1 value per channel when training, got input size torch.Size([1, 462])
EEGMiner: eval mode  → output (1, 4)
```

## Why this is confusing

A user who wants to run a single-sample forward (introspection, feature extraction, single-trial inference) reads the error message and looks for a model bug, when actually the model is in train mode by default and BatchNorm's behavior is correct given the mode.  All three models work with `batch_size >= 2` in train mode, and with any batch size in eval mode.

## Suggested fix (any of these)

**(a) Auto-`.eval()` at construction** — matches HuggingFace's convention; users explicitly opt into training with `.train()`.  Smallest semantic change for inference-first users.

**(b) Per-model `requires_min_batch_size` class attribute** + a friendlier error message that surfaces it.

**(c) Document in the docstrings** of the affected models that batch=1 forward requires `.eval()`.

Option (a) feels closest to the lightning / HF idiom and would silently fix this for all current and future BatchNorm-containing models without needing per-model tagging.

## Environment

- braindecode: $(pip show braindecode | head -2 | tail -1)
- PyTorch: 2.6
- Python 3.12

## Context

Found while smoke-testing a layer-wise probe feature across the full `models_dict` registry on neuralbench — these 3 models are the only ones in the registry where batch=1 forward fails in default mode.
